### PR TITLE
Profile page picture, bio, details, and users post added in the page

### PR DESCRIPTION
**Pull request by rekesan**

### CODE REVIEW

File: app/Login.tsx [hunk 1]
```diff
@@ -70,6 +71,15 @@ const SwrText = styled(Text)({
   fontWeight: "bold",
 });
 
+const setLoggedInUser = async (username: string) => {
+  try {
+    await AsyncStorage.setItem("username", username);
+    console.log(await AsyncStorage.getItem("username"))
```
Comment (jendan-nikolai) [inline app/Login.tsx]: remove unnecessary code

File: app/profile/[userId].tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+const getUserLoggedIn = async () => {
+  try {
+    return await AsyncStorage.getItem("username");
+  } catch (e) {
+    alert(e);
+  }
+};
+
+const getUser = async (p: { uName?: string; id?: number | string }) => {
```
Comment (jendan-nikolai) [inline app/profile/[userId].tsx]: be specific with the naming

**Review by jendan-nikolai — CHANGES_REQUESTED** [review]
be specific in naming variables and remove unused codes

File: app/_layout.tsx [hunk 3]
```diff
@@ -38,7 +39,11 @@ const Layout = () => {
     <DripsyProvider theme={theme}>
       <Stack
         screenOptions={({ route }) => ({
-          headerShown: route.name === "Profile" ? true : false,
+          headerShown: route.name.includes("profile") ? true : false,
```
Comment (jendan-nikolai) [inline app/_layout.tsx]: remove ternary condition

File: app/profile/[userId].tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+});
+
+const ProfileBioText = styled(Text)({ fontSize: 18, marginLeft: 15 });
+
+const Profile = () => {
+  const { userId } = useSearchParams();
+  const [user, setUser] = useState<UserProps>();
+  const sx = useSx();
+
+  if (userId === "me") {
```
Comment (jendan-nikolai) [inline app/profile/[userId].tsx:87]: avoid using abbreviations in variable names

File: service/data.tsx [hunk 5]
```diff
@@ -47,4 +56,33 @@ export const postData : PostProps[] = [
...
+    content:
+      "Jason didnt understand why his parents wouldnt let him sell his little sister at the garage sale." +
+      " He was 100% into fasting with her until he understood that meant he couldnt eat." +
+      "Just go ahead and press that button.",
+    imageURI: null,
+    reacts: { likes: 0 },
+  },
 ];
+
+export const mockDetails = [
```
Comment (mika-ella) [inline service/data.tsx]: create types for this

File: app/profile/[userId].tsx [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+
+import { UserProps } from "../../interface";
+import { mockDetails, postData, userData } from "../../service/data";
+import Button from "../../components/Button";
+import Post from "../../components/Post";
+
+const getUserLoggedIn = async () => {
+  try {
+    return await AsyncStorage.getItem("username");
+  } catch (e) {
```
Comment (mika-ella) [inline app/profile/[userId].tsx]: rename, what's the variabale e?

File: app/Login.tsx [hunk 7]
```diff
@@ -70,6 +71,15 @@ const SwrText = styled(Text)({
   fontWeight: "bold",
 });
 
+const setLoggedInUser = async (username: string) => {
+  try {
+    await AsyncStorage.setItem("username", username);
+    console.log(await AsyncStorage.getItem("username"))
+  } catch (e) {
```
Comment (mika-ella) [inline app/Login.tsx]: change variable name

File: app/profile/[userId].tsx [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+    alert(error);
+  }
+};
+
+const getUserData = async (props: {
+  username?: string;
+  id?: number | string;
+}) => {
+  return userData.find(
+    (user) => user.username === props.username || user.id == props.id
```
Comment (mika-ella) [inline app/profile/[userId].tsx]: add type for user

File: app/profile/[userId].tsx [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+
+const getUserLoggedIn = async () => {
+  try {
+    return await AsyncStorage.getItem("username");
+  } catch (error) {
+    alert(error);
+  }
+};
+
+const getUserData = async (props: {
```
Comment (mika-ella) [inline app/profile/[userId].tsx]: do not use props as a variable name, use an interface, might be userDTO

File: app/profile/[userId].tsx [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+
+  if (userId === "me") {
+    getUserLoggedIn().then((userLoggedIn) =>
+      getUserData({ username: userLoggedIn }).then((user) => setUser(user))
+    );
+  } else {
+    getUserData({ id: userId.toString() }).then((user) => setUser(user));
+  }
+
+  const mainButtonStyle = sx({
```
Comment (mika-ella) [inline app/profile/[userId].tsx]: rename, should be named based on its use

File: app/profile/[userId].tsx [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+    backgroundColor: "$primary",
+    width: "40%",
+    borderRadius: 5,
+  });
+
+  const secondButtonStyle = sx({
```
Comment (mika-ella) [inline app/profile/[userId].tsx]: do not use numbers to define a style

File: app/profile/[userId].tsx [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+    backgroundColor: "$light",
+    width: "40%",
+    borderRadius: 5,
+  });
+
+  const thirdButtonStyle = sx({
```
Comment (mika-ella) [inline app/profile/[userId].tsx]: do not use numbers to define a style

**Review by mika-ella — APPROVED** [review]

**Review by jendan-nikolai — APPROVED** [review]

**Pull request merged**